src/message/messageSelectors.js

```diff
@@ -18,7 +19,30 @@ import { NULL_ARRAY } from '../nullObjects';
 export const getPrivateMessages: Selector<Message[]> = createSelector(
   getAllNarrows,
   getMessages,
-  (narrows, messages) => (narrows[ALL_PRIVATE_NARROW_STR] || NULL_ARRAY).map(id => messages[id]),
+  (narrows, messages) => {
+    const privateMessages: Message[] = [];
+    const unknownIds: number[] = [];
+
+    const pmIds = narrows[ALL_PRIVATE_NARROW_STR] || NULL_ARRAY;
+    pmIds.forEach(id => {
+      const msg = messages[id];
+      if (msg !== undefined) {
+        privateMessages.push(msg);
+      } else {
+        unknownIds.push(id);
+      }
+    });
+
+    // BUG: all messages in `narrows` _should_ also be in `messages`.
+    // Error reports indicate that, somehow, this isn't always so.
```

Comment:
Interesting! And puzzling.

Is there an issue for this? This would be good to file an issue for.

Reply:
It would, and I should have filed one. Done, as #3749.

Reply:
Thanks!

Let's put that issue number in the comment, too.